springHomework3/src/main/java/org/example/CalculatorService.java

```diff
@@ -0,0 +1,45 @@
+package org.example;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class CalculatorService {
+    public static Calculator validCalculator(Calculator calculator) {
+        assert calculator != null;
```

Comment:
It's better to avoid `assert`. We coul throw RuntimeException or other Exception in this case